src/views/SystemIntake/index.tsx

```diff
@@ -58,46 +55,43 @@ export const SystemIntake = ({ match }: SystemIntakeProps) => {
     (state: AppState) => state.systemIntake.isLoading
   );
 
-  const renderPage = (formikProps: FormikProps<SystemIntakeForm>) => {
-    const Component = pageObj.view;
-
-    if (Component) {
-      return <Component formikProps={formikProps} />;
-    }
-    return null;
-  };
-
   const dispatchSave = () => {
     const currentRef = formikRef.current as FormikProps<SystemIntakeForm>;
     if (currentRef.dirty) {
       dispatch(saveSystemIntake(currentRef.values));
-      // Set initial values to those just saved so ref.dirty is compared against last saved values.
-      currentRef.resetForm({ values: currentRef.values });
-      if (!match.params.systemId) {
-        history.replace(`/system/${currentRef.values.id}`);
+      if (systemId === 'new') {
+        history.replace(`/system/${currentRef.values.id}/${pageObj.slug}`);
       }
     }
   };
 
   useEffect(() => {
-    if (match.params.systemId) {
-      dispatch(fetchSystemIntake(match.params.systemId));
-    } else {
+    if (systemId === 'new') {
       dispatch(storeSystemIntakeId(uuidv4()));
+    } else {
+      dispatch(fetchSystemIntake(systemId));
     }
-
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, []);
 
+  useEffect(() => {
+    const pageSlugs: any[] = pages.map(p => p.slug);
+    if (pageSlugs.includes(formPage)) {
```

Comment:
Instead of setting the page number in the `Route` `render`, I decided to set it here.

It seemingly worked fine, but React was complaining under the hood about setting the state in the render callback of the `Route` component.